core/lens/lens_operators.ml

```diff
@@ -0,0 +1,90 @@
+(* The operators named here are the ones that it is difficult or
+   impossible to define as "user" infix operators:
+
+      - -.  are both infix and prefix
+     && ||  have special evaluation
+     ::     is also used in patterns
+     ~      triggers a lexer state switch
+*)
+
+type name = string [@@deriving show]
+
+module Unary = struct
+  type t =
+    | Minus
+    | Not
+    | Name of name
+    [@@deriving show]
+
+  let from_links v =
+    match v with
+    | `Minus -> Minus
+    | `FloatMinus -> Minus
+    | `Not -> Not
+    | `Name name -> Name name
+
+  let to_string =
+  function
+    | Minus -> "-"
+    | Name name -> name
+    | Not -> "!"
+
+  let fmt f v =
+    Format.fprintf f "%s" (to_string v)
+end
+
+module Logical_binop = struct
+  type t =
+    | And
+    | Or
+    [@@deriving show]
+
+  let to_string =
+    function
+    | And -> "AND"
+    | Or -> "OR"
+
+  let fmt f v =
+    Format.fprintf f "%s" (to_string v)
+end
+
+module Binary = struct
+  type t =
+    | Minus
+    | Equal
+    | Cons
+    | Logical of Logical_binop.t
+    | Name of name
+    [@@deriving show]
+
+  let from_links v =
```

Comment:
Rather than `from`, I suggest you use `of`. Furthermore, the suffix `_links` is not really descriptive, as you are translating syntactic constructor in Links to another syntactic constructor in Links. Perhaps, something along the lines of `of_supertypes_operator` is more descriptive.